app/io/pathfinder/controllers/ClusterController.java

```
@@ -0,0 +1,104 @@
+package io.pathfinder.controllers;
+
+import com.avaje.ebean.Ebean;
+import com.avaje.ebean.text.json.JsonContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import javassist.compiler.SymbolTable;
+import play.libs.Json;
+import play.mvc.Controller;
+import io.pathfinder.models.Cluster;
+import play.mvc.Result;
+import java.util.Iterator;
+
+import javax.persistence.PersistenceException;
+
+public class ClusterController extends Controller {
+    private JsonContext jsonContext = Ebean.createJsonContext();
+
+    public Result getClusters() {
+        return ok(jsonContext.toJson(Cluster.find.all()));
+    }
+
+    public Result getCluster(long id) {
+        Cluster cluster = Cluster.find.byId(id);
+
+        if (cluster == null) {
+            return notFound();
+        }
+
+        return ok(jsonContext.toJson(cluster));
+    }
+
+    public Result createCluster() {
+        JsonNode json = request().body().asJson();
+        if (json == null) {
+            return badRequest("Expected content-type text/json or application/json");
+        }
+
+        Cluster cluster;
+
+        try {
+            cluster = Json.fromJson(json, Cluster.class);
+
+            cluster.save();
+            return created(jsonContext.toJson(cluster));
+        } catch (PersistenceException e) {
+            return internalServerError("Error saving cluster to the database: " + e.getMessage());
+        } catch (Exception e) {
```

Can we catch a more granular exception here? I don't think we should every catch all exceptions.

For example, if we catch a java.lang.InterruptedException it fucks with the JVM's scheduler.